Question_4.py

```diff
@@ -33,8 +33,27 @@ def fib_recursive(n):
         return answer
         
 print(n,'th Fibbonacci number', fib_recursive(n)[0]) # fib_recursive(n)[0] , here added [0] as the data is returned in the format of [fibo no, prev fibo no ]
+end = time.time()
+print(n,'Run time', end - start)
+#  on input 999 -> Run time 0.00040 seconds
 
 
+
+start = time.time()
+fib_values = {}
+def fib_recursive (num):
+    if num < 2 :                                 
+        return  num
```

Comment:
Remove extra space before `num`. Same for below `fib_recursive(num - 2 )` remove space after 2.

Please install black formatter and apply on each file. It will format the file correctly.